crates/script/src/lib.rs

```diff
@@ -248,10 +248,16 @@ impl ScriptArgs {
         let create2_deployer = state.script_config.evm_opts.create2_deployer;
         let compiled = state.compile()?;
 
+        // Validate that --verify is only used with --broadcast
+        if compiled.args.verify && !compiled.args.should_broadcast() {
+            eyre::bail!(
+                "The --verify flag requires --broadcast to be specified. Verification without broadcasting is not meaningful."
+            );
+        }
```

Comment:
you can do this with `requires` on `verify`: https://docs.rs/clap/latest/clap/struct.Arg.html#method.requires